package_bag/routines.py

```diff
@@ -26,7 +27,7 @@ def run(self):
             try:
                 bag_id = self.unpack_rename(bag, settings.TMP_DIR)
                 bag_path = join(settings.TMP_DIR, bag_id)
-                self.validate_structure(bag_path)
+                validate(bag_path)
                 self.validate_metadata(bag_path)
```

Comment:
I'm wondering if `validate_metadata` can return the bag info metadata, so you don't have to instantiate the bag again in `get_data`?

Reply:
I tried to implement this, but am running into issues where it's saving invalid bags to the database:
```bag_info = self.validate_metadata(bag_path)
                new_bag = Bag.objects.create(
                    original_bag_name=bag,
                    bag_identifier=bag_id,
                    bag_path=bag_path)
                self.get_data(bag, bag_info)```

Reply:
@bonniegee take a look at what I did L77-80

Reply:
Ah, okay--that makes sense